modules/mod-deep-learning/DeepLearningAnalyzer.cpp

```diff
@@ -0,0 +1,232 @@
+/**********************************************************************
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepLearningAnalyzer.cpp
+   Aldo Aguilar
+   Hugo Flores Garcia
+******************************************************************/
+/*
+\class DeepLearningAnalyzer
+\brief DeepLearningAnalyzer is an effect for labeling audio components in a track
+      using deep learning models.
+
+*/
+/*******************************************************************/
+
+#include "DeepLearningAnalyzer.h"
+
+#include <cstddef>
+#include <string>
+#include <vector>
+#include <wx/stattext.h>
+
+#include "FileNames.h"
+#include "Shuttle.h"
+#include "ShuttleGui.h"
+#include "WaveTrack.h"
+#include "LabelTrack.h"
+
+#include "effects/LoadEffects.h"
+
+#include <torch/script.h>
+
+// DeepLearningEffectBase implementation
+
+std::string DeepLearningAnalyzer::GetDeepEffectID()
+{ return "waveform-to-labels";}
+
+const ComponentInterfaceSymbol DeepLearningAnalyzer::Symbol
+{ XO("Deep Learning Analyzer") };
+
+// register audio Labeler
+namespace { BuiltinEffectsModule::Registration<DeepLearningAnalyzer> reg; }
+
+DeepLearningAnalyzer::DeepLearningAnalyzer() 
+{ 
+   SetLinearEffectFlag(false); 
+}
+
+DeepLearningAnalyzer::~DeepLearningAnalyzer() 
+{
+}
+
+
+// ComponentInterface implementation
+
+ComponentInterfaceSymbol DeepLearningAnalyzer::GetSymbol() { return Symbol; }
+
+TranslatableString DeepLearningAnalyzer::GetDescription() 
+{
+  return XO("The auto labeler uses deep learning models to "
+         "annotate audio tracks based on their contents automatically."); 
+}
+
+ManualPageID DeepLearningAnalyzer::ManualPage() 
+{
+  return L"Audio_Labeler"; 
+}
+
+// EffectDefinitionInterface implementation
+
+EffectType DeepLearningAnalyzer::GetType() { return EffectTypeAnalyze; }
+
+// Effect implementation
+
+// ProcessOne() takes a track, transforms it to bunch of buffer-blocks,
+// performs a forward pass through the deep model, and writes
+// the output to new tracks.
+bool DeepLearningAnalyzer::ProcessOne(WaveTrack *leader, double tStart, double tEnd) 
+{
+   // get current models labels
+   std::vector<std::string> classList = mModel->GetCard()->labels();
+   mClasses.clear();
```

Comment:
mClasses might remain not-clear when the analyzer exits, occupying a bit of memory needlessly.

But I think it is easily changed to a local variable instead of a class member variable, and passed as an argument to TensorToLabelTrack.